app/src/EditGroups.tsx

```diff
@@ -8,15 +13,180 @@ import "@blueprintjs/table/lib/css/table.css";
 import FileList from "./FileList";
 import { AxiosResponse } from "axios";
 
+export interface AddGroupRuleInput {
+  groupName: string;
+  ruleId: string;
+  ruleName: string;
+  ruleParams: string[];
+}
+
+export interface UpdateGroupNameInput {
+  oldGroupName: string;
+  newGroupName: string;
+}
+
+export interface UpdateRuleNameInput {
+  groupName: string;
+  oldRuleName: string;
+  newRuleName: string;
+}
+
+export type GroupAction =
+  | { type: "INIT_GROUPS"; groups: Group[] }
+  | { type: "NEW_GROUP" }
+  | { type: "REMOVE_GROUP"; groupName: string }
+  | { type: "ADD_GROUP_RULE"; addGroupRuleInput: AddGroupRuleInput }
+  | { type: "UPDATE_GROUP_NAME"; updateGroupNameInput: UpdateGroupNameInput }
+  | { type: "UPDATE_RULE_NAME"; updateRuleNameInput: UpdateRuleNameInput };
+
+export abstract class GroupActions {
+  static initGroupAction(groups: Group[]): GroupAction {
+    return { type: "INIT_GROUPS", groups };
```

Comment:
Consider moving the hard-coded constants (e.g. "INIT_GROUPS") to declared constants... will help if you next decide to move the actions, reducers and UI to separate files

Reply:
Good point. Will change nx pr